concrete/src/File/Image/Thumbnail/Path/Resolver.php

```diff
@@ -111,6 +115,35 @@ protected function getStoredPath($file_id, $version_id, $storage_location_id, $t
     }
 
     /**
+     * Checks whether or not a thumbnail exists in the file storage location for this file and
+     * returns the passed string if the thumbnail exists or the URL of the file if it does not.
+     *
+     * In the case of non-image files with thumbnails, if the thumbnail file does not exist it returns null.
+     *
+     * @param string $path
+     * @param \Concrete\Core\Entity\File\Version $fileVersion
+     * @param \Concrete\Core\File\Image\Thumbnail\Type\Version $thumbnail
+     * @param \Concrete\Core\Entity\File\StorageLocation\StorageLocation $storageLocation
+     * @return mixed
+     */
+    protected function checkPathExists($path, Version $fileVersion, ThumbnailVersion $thumbnail, StorageLocation $storageLocation)
+    {
+
+        $fileSystem = $storageLocation->getFileSystemObject();
+
+        if (!$fileSystem->has($thumbnail->getFilePath($fileVersion))) {
+
+            if ($fileVersion->getTypeObject()->getGenericType() == FileType::T_IMAGE) {
+                $path = $fileVersion->getURL();
```

Comment:
This breaks BC doesn't it? and we do caching at the storage adapter level to say if files exist or not, the existing solution can have issues if the files are deleted from the filesystem without c5 knowing.